onnx/test/cpp/function_context_test.cc

```diff
@@ -149,7 +149,265 @@ void RegisterCustomFunctionSchema() {
+                    output = Sub (X_Sub, X_Log)
+                )");
+            schema.BuildFunction(functionProto);
+            return true;
+          },
+          13);
+  std::cout << "StepB\n"; 
+  if (!with_missing_version) {
+    schema.SetContextDependentFunctionBodyBuilder(
+        [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) -> bool {
+          const int64_t axis = ctx.getAttribute("axis") != nullptr ? ctx.getAttribute("axis")->i() : -1;
+          FunctionBuilder builder(functionProto);
+          builder.Const1D("axes", axis).Add("X_ReduceMax = ReduceMax <keepdims = 1> (input, axes)").Add(R"(
+                    X_Sub = Sub (input, X_ReduceMax)
+                    X_Exp = Exp (X_Sub)
+                    X_ReduceSum = ReduceSum <keepdims = 1> (X_Exp, axes)
+                    X_Log = Log (X_ReduceSum)
+                    output = Sub (X_Sub, X_Log)
+                )");
+          schema.BuildFunction(functionProto);
+          return true;
+        },
+        18);
+  }
+  std::cout << "StepD\n"; 
+  ONNX_NAMESPACE::OpSchemaRegistry::OpSchemaRegisterOnce unused(schema);
+  (void)unused;
+}
+
+void BuildLogSoftmaxFunction(
+    const OpSchema& schema,
+    const std::string& op_type,
+    FunctionProto& fnProto,
+    int requested_opset_version) {
+  NodeProto nodeProto;
+  nodeProto.set_op_type(op_type);
+  nodeProto.add_input("input");
+  nodeProto.add_output("output");
+
+  TypeProto floatTypeProto;
+  floatTypeProto.mutable_tensor_type()->set_elem_type(TensorProto_DataType::TensorProto_DataType_FLOAT);
+
+  FunctionBodyBuildContextImpl ctx(nodeProto, {floatTypeProto});
+  schema.BuildContextDependentFunctionWithOpsetVersion(ctx, fnProto, requested_opset_version, "", true);
+}
+
+TEST(FunctionAPITest, VersionedFunctionWithMissingVersionTest) {
+  std::string op_type = "VersionedLogSoftMax";
+  std::cout << "StepA\n"; 
+  RegisterVersionedLogSoftmaxFunctionSchema(op_type, true);
+
+  // The function op's since version is 13. there is no new version after version 13.
+  // As a result, all OpSchema shall point to the same one.
+  const auto* schema13 = OpSchemaRegistry::Schema(op_type, 13, ONNX_DOMAIN);
+  EXPECT_TRUE(schema13);
+  const auto* schema17 = OpSchemaRegistry::Schema(op_type, 17, ONNX_DOMAIN);
+  EXPECT_TRUE(schema17 == schema13);
+  const auto* schema18 = OpSchemaRegistry::Schema(op_type, 18, ONNX_DOMAIN);
+  EXPECT_TRUE(schema18 == schema13);
+
+  std::cout << "Step1\n"; 
+  // VersionedLogSoftMax has 1 function body for opset version 13.
+  // It is ok to request a function body for a model with opset import version 13.
+  FunctionProto function_proto13;
+  BuildLogSoftmaxFunction(*schema13, op_type, function_proto13, 13);
+  ASSERT_TRUE(function_proto13.name() == op_type);
+  std::cout << "Step2\n"; 
+
+  ONNX_TRY {
+    // It may not be ok to request a function body for a model with opset import version 14, 15, 16, and 17.
```

Comment:
nit: suggest changing the wording to `A request for a function body .... 14, 15, 16, and 17 will fail`